soroban-ledger-snapshot/src/lib.rs

```diff
@@ -118,6 +124,61 @@ impl LedgerSnapshot {
             }
         }
     }
+
+    pub fn add_contract_to_entries(&mut self, contract_id: [u8; 32], wasm_hash: [u8; 32]) {
+        let entries = &mut self.ledger_entries;
+        // Create the contract
+        let contract_key = LedgerKey::ContractData(LedgerKeyContractData {
+            contract_id: contract_id.into(),
+            key: ScVal::Static(ScStatic::LedgerKeyContractCode),
+        });
+
+        let contract_entry = LedgerEntry {
+            last_modified_ledger_seq: 0,
+            data: LedgerEntryData::ContractData(ContractDataEntry {
+                contract_id: contract_id.into(),
+                key: ScVal::Static(ScStatic::LedgerKeyContractCode),
+                val: ScVal::Object(Some(ScObject::ContractCode(ScContractCode::WasmRef(Hash(
+                    wasm_hash,
+                ))))),
+            }),
+            ext: LedgerEntryExt::V0,
+        };
+        for (k, e) in entries.iter_mut() {
+            if **k == contract_key {
+                **e = contract_entry;
+                return;
+            }
+        }
+        entries.push((Box::new(contract_key), Box::new(contract_entry)));
+    }
+
+    pub fn add_contract_code_to_entries(&mut self, contract: &[u8]) -> Result<Hash, xdr::Error> {
+        let entries = &mut self.ledger_entries;
+        let args_xdr = InstallContractCodeArgs {
+            code: contract.try_into()?,
+        }
+        .to_xdr()?;
+        let hash = Hash(sha2::Sha256::digest(args_xdr).into());
+        let code_key = LedgerKey::ContractCode(LedgerKeyContractCode { hash: hash.clone() });
+        let code_entry = LedgerEntry {
+            last_modified_ledger_seq: 0,
+            data: LedgerEntryData::ContractCode(ContractCodeEntry {
+                code: contract.try_into()?,
+                ext: ExtensionPoint::V0,
+                hash: hash.clone(),
+            }),
+            ext: LedgerEntryExt::V0,
+        };
+        for (k, e) in entries.iter_mut() {
+            if **k == code_key {
+                **e = code_entry;
```

Comment:
Could you add happy path tests for both new functions, given they have some interesting logic around sha2-ing data?